app/assets/javascripts/analytics.js

```diff
@@ -1,190 +1,164 @@
-GoogleAnalytics = (function() {
-  function GoogleAnalytics() {}
-
-  GoogleAnalytics.load = function() {
-    GoogleAnalytics.analyticsId = GoogleAnalytics.getAnalyticsId();
-    (function(i, s, o, g, r, a, m) {
-      i['GoogleAnalyticsObject'] = r;
-      i[r] = i[r] || function() {
-        (i[r].q = i[r].q || []).push(arguments);
-      };
-
-      i[r].l = 1 * new Date;
-
-      a = s.createElement(o);
-      m = s.getElementsByTagName(o)[0];
-
-      a.async = 1;
-      a.src = g;
-      m.parentNode.insertBefore(a, m);
-    })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
-    ga('create', GoogleAnalytics.analyticsId, 'auto');
-    ga('set', 'anonymizeIp', true);
-  };
-
-  GoogleAnalytics.trackPageview = function(url) {
-    if (!GoogleAnalytics.isLocalRequest()) {
-      return ga('send', {
-          hitType: 'pageview',
-          page: GoogleAnalytics.getPath()
-        }
-      );
-    }
-  };
-
-  GoogleAnalytics.isLocalRequest = function() {
-    return GoogleAnalytics.documentDomainIncludes('local');
-  };
-
-  GoogleAnalytics.documentDomainIncludes = function(str) {
-    return document.domain.indexOf(str) !== -1;
-  };
-
-  GoogleAnalytics.getAnalyticsId = function() {
-    return $('[data-analytics-id]').data('analytics-id');
-  };
-
-  // Remove the protocol and the host and only return the path with any params
-  GoogleAnalytics.getPath = function() {
-    return location.href
-             .replace(location.protocol + '//' + location.host, '');
-  };
-
-  return GoogleAnalytics;
-
-})();
-
-$(document).on('turbolinks:load', function(){
-  GoogleAnalytics.load();
-  if (GoogleAnalytics.analyticsId){
-    GoogleAnalytics.trackPageview();
-  }
-
-  $('.btn-renewable-submit').on('click', function(e) {
-    ga('send', 'event', {
-      eventCategory: 'Renew',
-      eventAction: 'renew-single',
-      transport: 'beacon'
-    });
+// gtag intial setup
+window.dataLayer = window.dataLayer || [];
+function gtag(){dataLayer.push(arguments);}
+gtag('js', new Date());
+
+// Send event in required GA4 format to Google
+function sendAnalyticsEvent({ action, category, label, value }) {
+  window.gtag && window.gtag('event', action, {
+    event_category: category,
+    event_label: label,
+    event_value: value
+  });  
+}
+
+document.addEventListener('turbolinks:load', function() {
```

Comment:
does this fire on the very first page load? like, if I visit the site and do nothing else, will it still run and enable debug mode?

Reply:
@thatbudakguy and I confirmed that the events are sending on the first page load.